tests/saitests/sai_qos_tests.py

```diff
@@ -579,6 +580,112 @@ def runTest(self):
         finally:
             print >> sys.stderr, "END OF TEST"
 
+
+# Tunnel DSCP to PG mapping test
+class TunnelDscpToPgMapping(sai_base_test.ThriftInterfaceDataPlane):
+
+    def _build_testing_pkt(self, active_tor_mac, standby_tor_mac, active_tor_ip, standby_tor_ip, inner_dscp, outer_dscp, dst_ip, ecn=1):
+        pkt = simple_tcp_packet(
+                eth_dst=standby_tor_mac,
+                ip_src='1.1.1.1',
+                ip_dst=dst_ip,
+                ip_dscp=inner_dscp,
+                ip_ecn=ecn,
+                ip_ttl=64
+                )
+        
+        ipinip_packet = simple_ipv4ip_packet(
+                            eth_dst=active_tor_mac,
+                            eth_src=standby_tor_mac,
+                            ip_src=standby_tor_ip,
+                            ip_dst=active_tor_ip,
+                            ip_dscp=outer_dscp,
+                            ip_ecn=ecn,
+                            inner_frame=pkt[scapy.IP]
+                            )
+        return ipinip_packet
+
+    def runTest(self):
+        """
+        This test case is to tx some ip_in_ip packet from Mux tunnel, and check if the traffic is
+        mapped to expected PGs.
+        """
+        switch_init(self.client)
+
+        # Parse input parameters
+        active_tor_mac = self.test_params['active_tor_mac']
+        active_tor_ip = self.test_params['active_tor_ip']
+        standby_tor_mac = self.test_params['standby_tor_mac']
+        standby_tor_ip = self.test_params['standby_tor_ip']
+        src_port_id = self.test_params['src_port_id']
+        src_port_slice = self.test_params['src_port_slice']
+        dst_port_id = self.test_params['dst_port_id']
+        dst_port_ip = self.test_params['dst_port_ip']
+
+        dscp_to_tc_map = self.test_params['tunnel_qos_map']['dscp_to_tc_map']
+        tc_to_pg_map = self.test_params['tunnel_qos_map']['tc_to_priority_group_map']
+        asic_type = self.test_params['sonic_asic_type']
+        dscp_to_pg_map = {}
+        PKT_NUM = 100
+        # We noticed that a packet less than 256 bytes will occupy a 256-sized buffer
+        FACTOR = 256
```

Comment:
this will be different for Th2 since the cell size is 208. for TD3, the cell size is 256.

Reply:
Fixed. Thanks